packages/flutter/lib/src/cupertino/scrollbar.dart

```diff
@@ -0,0 +1,106 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/widgets.dart';
+
+// All values eyeballed.
+const Color _kScrollbarColor = const Color(0x99777777);
+const double _kScrollbarThickness = 2.5;
+const double _kScrollbarMainAxisMargin = 4.0;
+const double _kScrollbarCrossAxisMargin = 2.5;
+const double _kScrollbarMinLength = 4.0;
+const Radius _kScrollbarRadius = const Radius.circular(1.25);
+const Duration _kScrollbarTimeToFade = const Duration(milliseconds: 50);
+
+/// A iOS style scrollbar.
+///
+/// A scrollbar indicates which portion of a [Scrollable] widget is actually
+/// visible.
+///
+/// To add a scrollbar to a [ScrollView], simply wrap the scroll view widget in
+/// a [CupertinoScrollbar] widget.
+///
+/// See also:
+///
+///  * [ListView], which display a linear, scrollable list of children.
+///  * [GridView], which display a 2 dimensional, scrollable array of children.
+///  * [Scrollbar], a Material Design scrollbar that dynamically adapts to the
+///    platform showing either an Android style or iOS style scrollbar.
+class CupertinoScrollbar extends StatefulWidget {
+  /// Creates an iOS style scrollbar that wraps the given [child].
+  ///
+  /// The [child] should be a source of [ScrollNotification] notifications,
+  /// typically a [Scrollable] widget.
+  const CupertinoScrollbar({
+    Key key,
+    @required this.child,
+  }) : super(key: key);
+
+  /// The subtree to place inside the [CupertinoScrollbar].
+  ///
+  /// This should include a source of [ScrollNotification] notifications,
+  /// typically a [Scrollable] widget.
+  final Widget child;
+
+  @override
+  _CupertinoScrollbarState createState() => new _CupertinoScrollbarState();
+
+  static ScrollbarPainter buildCupertinoScrollbarPainter(TickerProvider vsync) {
```

Comment:
Would it make sense for this just to be a private member on the state?

Reply:
Material scrollbar actually draws with cupertino painter on iOS. Seems consistent with googlekit on iOS.

Reply:
ah, got it.